src/meltano/cli/cli.py

```diff
@@ -47,6 +47,11 @@ def main(self, *args, **kwargs) -> t.NoReturn:
     cls=NoWindowsGlobbingGroup,
     invoke_without_command=True,
     no_args_is_help=True,
+    # For backwards compatibility, accept CLI options that use underscores
+    # instead of hyphens.
+    # https://github.com/pallets/click/issues/1123#issuecomment-589989721
+    # NOTE: This CLI option normalization applies to all subcommands.
+    context_settings={"token_normalize_func": lambda x: x.replace("_", "-")},
```

Comment:
This *will* lock us out of intentionally using underscores in option names in the future, but I don't expect us to ever have a need to distinguish between 2 options whose only differentiators are underscores vs hyphens